python/ray/_private/services.py

```diff
@@ -919,7 +919,23 @@ def start_redis(node_ip_address,
         # If no port is given, fallback to default Redis port for the primary
         # shard.
         if port is None:
-            port = ray_constants.DEFAULT_PORT
+            default_port = ray_constants.DEFAULT_PORT
+            if sys.platform == "win32":
+                port = default_port
+                num_choices = 200
+                a_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                while num_choices > 0:
+                    location = (node_ip_address, port)
+                    try:
+                        a_socket.bind(location)
+                        break
+                    except socket.error as e:
+                        if e.errno == errno.EADDRINUSE:
```

Comment:
Don't forget to reraise or handle the exception if the errno is something else.

Reply:
Yes. Just changing port in case of any error in `a_socket.bind` sounded simple to me. Because, anyways, out of 200 ports at least one will work.